client/src/modules/users/UserDepotManagementModal.html

```diff
@@ -19,14 +19,37 @@ <h4><i> {{ UsersDepotModalCtrl.user.display_name }} </i></h4>
     </div>
 
     <span translate> FORM.INFO.USER_DEPOT </span>
+    <div class="checkbox">
+      <label data-root-node>
+        <input type="checkbox" ng-model="UsersDepotModalCtrl.all" ng-change="UsersDepotModalCtrl.setAllNodeValue(UsersDepotModalCtrl.depotsData, UsersDepotModalCtrl.all)" />
+        <strong class="text-capitalize" translate>FORM.LABELS.CHECK_ALL</strong>
+      </label>
+    </div>
+    
+    <div class="panel panel-default" style="margin-bottom : 0px;">
+      <div class="panel-heading clearfix" style="text-align : left">
+        <a href ng-click="UsersDepotModalCtrl.toggleFilter()">
+          <span translate>TABLE.COLUMNS.FILTERING.LABEL</span> <span class="caret" ng-class="{'rotate' : UsersDepotModalCtrl.filterActive}"></span>
+        </a>
+      </div>
 
-    <bh-depot-search-select
-      label="STOCK.DEPOT"
-      id="user_depots"
-      depots-uuids = "UsersDepotModalCtrl.depots"
-      on-change="UsersDepotModalCtrl.onDepotChange(depots)">
-    </bh-depot-search-select>
+      <input ng-model="UsersDepotModalCtrl.filter" ng-show="UsersDepotModalCtrl.filterActive" class="form-control" placeholder="{{'FORM.PLACEHOLDERS.FILTER_NAME' | translate }}" style="border-radius: 0"/>
+    </div>
 
+    <div ng-if="!UsersDepotModalCtrl.loading" ng-repeat="depot in UsersDepotModalCtrl.depotsData | filter : { text : UsersDepotModalCtrl.filter }">
+      <div style="margin-left: calc({{depot.$$treeLevel}} * 15px)" class="list-unstyled">
+        <div class="checkbox">
+          <label data-label="{{child._label}}">
+            <input type="checkbox" id="depot_{{$index}}" ng-model="depot._checked" ng-change="UsersDepotModalCtrl.setNodeValue(depot.children, depot)" />
```

Comment:
Instead of `id=depot_{{$index}}`, why don't use use the depot uuid?  That way, it is guaranteed to be unique and you can refer to the depot by its UUID directly.

Reply:
It was only a trick with for the **end to end** test

Reply:
As mentioned in person, if you could use an identifier from the depot, rather than it's placement in the array, this will save a lot of test errors in the future.